dev-tools/omdb/src/bin/omdb/reconfigurator.rs

```diff
@@ -192,3 +224,125 @@ async fn cmd_reconfigurator_archive(
 
     Ok(())
 }
+
+/// Show recent history of blueprints
+async fn cmd_reconfigurator_history(
+    opctx: &OpContext,
+    datastore: &DataStore,
+    history_args: &HistoryArgs,
+) -> anyhow::Result<()> {
+    // Select recent targets.
+    let limit = history_args.limit;
+    let mut targets: Vec<_> = {
+        use nexus_db_queries::db::schema::bp_target::dsl;
+        let conn = datastore
+            .pool_connection_for_tests()
+            .await
+            .context("obtaining connection")?;
+        dsl::bp_target
+            .select(BpTarget::as_select())
+            .order_by(dsl::version.desc())
+            .limit(i64::from(limit))
+            .get_results_async(&*conn)
+            .await
+            .context("listing targets")?
+    };
+
+    // Select everything from the blueprint table.
+    // This shouldn't be very large.
+    let mut all_blueprints: BTreeMap<BlueprintUuid, BlueprintMetadata> =
+        BTreeMap::new();
+    let mut paginator = Paginator::new(SQL_BATCH_SIZE);
+    while let Some(p) = paginator.next() {
+        let records_batch = datastore
+            .blueprints_list(opctx, &p.current_pagparams())
+            .await
+            .context("batch of blueprints")?;
+        paginator = p.found_batch(&records_batch, &|b| *b.id.as_untyped_uuid());
+        all_blueprints.extend(records_batch.into_iter().map(|b| (b.id, b)));
+    }
+
+    // Sort the target list in increasing order.
+    // (This should be the same as reversing it.)
+    targets.sort_by_key(|b| b.version);
+
+    // Now, print the history.
+    println!("{:>5} {:24} {:36}", "VERSN", "TIME", "BLUEPRINT");
+    if targets.len() == usize::try_from(limit).unwrap() {
+        println!("... (earlier history omitted)");
+    }
+
+    let mut prev: Option<BlueprintUuid> = None;
+    let mut prev_blueprint: Option<Blueprint> = None;
+    for t in targets {
+        let target_id = BlueprintUuid::from(t.blueprint_id);
+        let this_previous = prev_blueprint.take();
+        prev_blueprint = None;
+
+        print!(
+            "{:>5} {} {} {:>8}",
+            t.version,
+            humantime::format_rfc3339_millis(t.time_made_target.into()),
+            target_id,
+            if t.enabled { "enabled" } else { "disabled" },
+        );
+
+        let same_blueprint = matches!(prev,
+            Some(prev_id) if prev_id == target_id
+        );
```

Comment:
I think `matches!` is overkill; could be

```suggestion
        let same_blueprint = prev == Some(target_id);
```